libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/dh-market-participant-actor-market-roles.component.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { DhApiModule } from '@energinet-datahub/dh/shared/data-access-api';
+import { MarketRoleService } from './market-role.service';
+import {
+  EicFunction,
+  MarketRoleDto,
+} from '@energinet-datahub/dh/shared/domain';
+
+describe('MarketRolesChk', () => {
+  async function setup(marketRolesArrToTest: MarketRoleDto[]) {
+    return await render(DhMarketParticipantActorMarketRolesComponent, {
+      componentProperties: {
+        actor: {
+          actorId: '123',
+          externalActorId: '1234',
+          gln: { value: '13' },
+          gridAreas: [],
+          marketRoles: marketRolesArrToTest,
+          meteringPointTypes: [],
+          status: 'New',
+        },
+      },
+      imports: [
+        HttpClientModule,
+        DhApiModule.forRoot(),
+        getTranslocoTestingModule(),
+      ],
+    });
+  }
+
+  function getAllOptions() {
+    return screen.getAllByRole('option');
+  }
+
+  test('should render checkbox list', async () => {
+    await setup([]);
+
+    const elm = screen.getByRole('listbox');
+    expect(elm).toBeInTheDocument();
+  });
+
+  test('should render checkbox options', async () => {
+    await setup([]);
+
+    const marketRoleService = new MarketRoleService();
+    const availableMarketRolesCount =
+      marketRoleService.getAvailableMarketRoles.length;
+
+    const allOptions = getAllOptions();
+    expect(allOptions).toHaveLength(availableMarketRolesCount);
+  });
+
+  test('should render checkbox options, preselected', async () => {
+    const view = await setup([{ eicFunction: EicFunction.GridAccessProvider }]);
+
+    await view.fixture.whenStable();
+
+    view.fixture.detectChanges();
+
+    const allOptions = getAllOptions();
+    const selectedRoles = allOptions.filter(
+      (e) => e.attributes.getNamedItem('aria-selected')?.value == 'true'
+    );
+    expect(selectedRoles).toHaveLength(1);
+  });
+
+  test('should render checkbox options, on selection', async () => {
+    // Arrange
+    const view = await setup([]);
+
+    await view.fixture.whenStable();
+
+    const allOptions = getAllOptions();
+
+    const elemToSelect1 = allOptions[0] as HTMLOptionElement;
+    const elemToSelect2 = allOptions[1] as HTMLOptionElement;
+    const elemNotSelectable = allOptions[5] as HTMLOptionElement;
+
+    // Act
+    userEvent.click(elemToSelect1);
+    userEvent.click(elemToSelect2);
+    userEvent.click(elemNotSelectable, undefined, {
+      skipPointerEventsCheck: true,
+    }); // not selectable by disabled validation
+
+    view.fixture.detectChanges();
+
+    // Assert
+    const selectedRoles = allOptions.filter(
+      (e) => e.attributes.getNamedItem('aria-selected')?.value == 'true'
+    );
+    const changes = view.fixture.componentInstance.changes;
+
+    expect(selectedRoles).toHaveLength([elemToSelect1, elemToSelect2].length);
+    expect(changes.marketRoles).toHaveLength(
+      [elemToSelect1, elemToSelect2].length
+    );
+  });
```

Comment:
suggestion: it'd be nice to have at least one test that covers the output functionality. It's an important functionality for this component.